src/secret_sharing/mod.rs

```diff
@@ -11,18 +11,20 @@ use crate::ff::ArithmeticOps;
 use std::fmt::Debug;
 
 pub trait SharedValue:
-    Clone + Copy + PartialEq + Debug + Send + Sync + Sized + Serializable + 'static
+    Clone
+    + Copy
+    + PartialEq
+    + Debug
+    + Send
+    + Sync
+    + Sized
+    + ArithmeticOps
+    + BooleanOps
```

Comment:
I think that you might need to add one extra layer to this, based on the above comments on fields.

That is, we include `ArithmeticOps` for all, but we need an additional constraint for `Fp2`.

If we ever move to a ring in $F_{2^n}$, we'll be in a position to use the binary operations, but we can't do that in a generic field.

Reply:
If we ever actually need Fp2 field values to form `Fp2Array`, I'll add an extra layer to for boolean operations. For now, I added `BooleanOps` bound directly to `Fp2Array`.

Here's the new diagram.

![image](https://user-images.githubusercontent.com/8149758/218390250-e8ae220c-5090-4b75-892f-223716f30cec.png)

Reply:
Works for me, thanks.